lisa/tools/fio.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import TYPE_CHECKING, List, Type, cast
+
+from lisa.executable import Tool
+from lisa.operating_system import Posix
+
+from .git import Git
+
+if TYPE_CHECKING:
+    from .make import Make
+
+
+class Fio(Tool):
+    fio_repo = "https://github.com/axboe/fio/"
+    branch = "fio-3.28"
+
+    @property
+    def command(self) -> str:
+        return "fio"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Git, Make]
+
+    def _install_from_src(self) -> bool:
+        tool_path = self.get_tool_path()
+        self.node.shell.mkdir(tool_path, exist_ok=True)
+        git = self.node.tools[Git]
+        git.clone(self.fio_repo, tool_path)
+        code_path = tool_path.joinpath("fio")
+        git.checkout(
+            ref="refs/heads/master", cwd=code_path, checkout_branch=self.branch
+        )
+        make = self.node.tools[Make]
+        make.make_install(cwd=code_path)
+        return self._check_exists()
+
+    def install(self) -> bool:
+        posix_os: Posix = cast(Posix, self.node.os)
+        posix_os.install_packages("fio")
+        if not self._check_exists():
+            self._install_from_src()
+        return self._check_exists()
+
+    def launch(
+        self,
+        name: str,
+        filename: str,
+        mode: str,
+        runtime: int,
+        iodepth: int,
+        numjob: int,
+        block_size: str = "4K",
+        direct: bool = True,
+        gtod_reduce: bool = True,
+        ioengine: str = "libaio",
+        group_reporting: bool = True,
+    ) -> None:
+        cmd = (
+            f"--ioengine={ioengine} --bs={block_size} --filename={filename} "
+            f"--readwrite={mode} --runtime={runtime} --iodepth={iodepth} "
+            f"--numjob={numjob} --name={name}"
+        )
+        if direct:
+            cmd += " --direct=1"
+        if gtod_reduce:
+            cmd += " --gtod_reduce=1"
+        if group_reporting:
+            cmd += " --group_reporting"
+        self.run(
+            cmd,
+            force_run=True,
+            sudo=True,
```

Comment:
Does it need sudo?

Reply:
Launch fio don't need sudo, but when run against device, it need sudo, I tried.